qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1126,8 +1126,13 @@ def substitute_node_with_dag(self, node, input_dag, wires=None, propagate_condit
                 )
 
         reverse_wire_map = {b: a for a, b in wire_map.items()}
-        op_condition = getattr(node.op, "condition", None)
-        if propagate_condition and op_condition is not None:
+        # It doesn't make sense to try and propagate a condition from a control-flow op; a
+        # replacement for the control-flow op should implement the operation completely.
```

Comment:
Do you think we should document this? I think it's unlikely to come up in practice, because most of the use cases of `substitute_node_with_dag` will handle the recursive nesting outside of this.

I guess the larger around this is also how do you handle clbit mapping with the replacement if you did do the substitution.

Reply:
I can do - is a small note in the docstring enough?

For propagation: if we allowed condition propagation from (say) `IfElseOp` to the replacement DAG, we'd effectively be limiting the only usable substitution to replacing an `IfElseOp` with no "false" block with exactly the content of the "true" block (or some simplification of that).  It's not really a logical operations.

Fwiw, I hope that the current system of control-flow representation inside `DAGCircuit` goes away, in which case control-flow ops won't be nodes that can be substituted at all.

Reply:
Yeah a note in the docstring is probably enough.

But now I'm wondering if we want to just error if `node` is a `ControlFlowOp` if there isn't a potential use case where it would work. I don't think we ever call it on a `ControlFlowOp` and maybe we should make it explicit that it doesn't make any sense?

Reply:
The substitution works fine without condition propagation, and I can certainly imagine situations where that could be useful.

Reply:
I mean, I don't think _Qiskit_ will ever do that, but I think other people trying to see what's possible with dynamic circuits and building on top of Qiskit might want to do it.

Reply:
Ok, then just a note in the docstring explaining how this works for `ControlFlowOp` I think would be good to have here

Reply:
Added a docstring comment in 22763810c.